src/riak_core_info_service_process.erl

```diff
@@ -0,0 +1,113 @@
+-module(riak_core_info_service_process).
```

Comment:
I was thinking the same thing - simple `gen_server` with a few `handle_info` function heads should work just as well, and be less boilerplate code. I didn't know what the original reasoning was - maybe @macintux can weigh in and if he agrees we can just turn this into a `gen_server` process.